tests/e2e/bento_server_http/tests/test_serve.py

```diff
@@ -88,3 +92,17 @@ def test_serve_from_svc():
         # the process, it should be negative.
         # on all other platforms, this should be 0.
         assert server.process.poll() <= 0
+
+
+def test_serve_with_timeout(bentoml_home: str):
+    server = bentoml.HTTPServer("service.py:svc", port=12346)
+    config_file = os.path.abspath("configs/timeout.yml")
+    env = os.environ.copy()
+    env.update(BENTOML_CONFIG=config_file)
+
+    with server.start(env=env) as client:
+        with pytest.raises(
+            BentoMLException,
+            match="504: b'Not able to process the request in 1 seconds'",
+        ):
+            client.echo_delay()
```

Comment:
Let's update the config.yml to `server_config_file` fixture under this conftest

Reply:
The fixture is a matrix with other config files not related to the test. Should we update it?

Reply:
I think it is fine just to do what you are doing.